tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -267,20 +809,29 @@ function Use-AzureRmProfile
     $ProfileMap = (Get-AzProfile)
     $AllProfiles = ($ProfileMap | Get-Member -MemberType NoteProperty).Name
     $Profile = $PSBoundParameters.Profile
+    $Scope = $PSBoundParameters.Scope
+    $Modules = $PSBoundParameters.Module
+
+    if ($Modules -eq $null)
+    {
+      $Modules = ($ProfileMap.$Profile | Get-Member -MemberType NoteProperty).Name
+    }
+
     if ($PSCmdlet.ShouldProcess($Profile, "Loading modules for profile in the current scope"))
     {
-      $version = Get-AzureRmModule -Profile $Profile -Module $RollUpModule
-      if (($version -eq $null) -and ($Force.IsPresent -or $PSCmdlet.ShouldContinue("Install Modules for Profile $Profile from the gallery?", "Installing Modules for Profile $Profile")))
+      Write-Host "Loading Profile $Profile"
+      foreach ($Module in $Modules)
       {
-        $versions = $ProfileMap.$Profile.$RollUpModule
-        $versionEnum = $versions.GetEnumerator()
-        $toss = $versionEnum.MoveNext()
-        $version = $versionEnum.Current
-        Install-Module $RollUpModule -RequiredVersion $version -Repository $BootStrapRepo
+        $version = Get-AzureRmModule -Profile $Profile -Module $Module
+        if (($version -eq $null) -and ($IsConfirmed -or $Force.IsPresent -or $PSCmdlet.ShouldContinue("Install Module $module for Profile $Profile from the gallery?", "Installing Modules for Profile $Profile")))
```

Comment:
We should make use of the yes to all / no to all prompt here (https://msdn.microsoft.com/en-us/library/mt659128(v=vs.85).aspx) , rather than adding the $IsConfirmed boolean.

I also think we should guard each of these with a separate ShouldProcess prompt, so the user can find out which modules would be installed from the gallery using -WhatIf